src/anari_test_scenes/scenes/file/gltf2anari.h

```diff
@@ -462,56 +710,260 @@ struct gltf_data
+                  bc7 = true;
+                }
+              }
+            }
+            bool needsTranscoding = ktxTexture2_NeedsTranscoding(texture);
+            if (bc7 && needsTranscoding) {
+              result = ktxTexture2_TranscodeBasis(texture, KTX_TTF_BC7_RGBA, 0);
+
+              if (result != KTX_SUCCESS) {
+                printf("Failed to load image %zu: %s\n", imgIndex,
+                    ktxErrorString(result));
+              }
+
+              *width = texture->baseWidth;
+              *height = texture->baseHeight;
+              *n = ktxTexture2_GetNumComponents(texture);
+              int fmt = texture->vkFormat;
+
+              result = ktxTexture_GetImageOffset(
+                  ktxTexture(texture), 0, 0, 0, &offset);
+
+              if (result != KTX_SUCCESS) {
+                printf("Failed to load image %zu: %s\n", imgIndex, ktxErrorString(result));
+              }
+
+              data = ktxTexture_GetData(ktxTexture(texture)) + offset;
+              int bytes = ktxTexture_GetDataSize(ktxTexture(texture));
+              json samplerExtras = json::object();
+              samplerExtras["type"] = "compressedImage2D";
+              samplerExtras["bytes"] = bytes;
+              samplerExtras["format"] = "BC7";
+              samplerExtras["width"] = texture->baseWidth;
+              samplerExtras["height"] = texture->baseHeight;
+
+              for (auto &texture : gltf.value("textures", json::object())) {
+                int ktxImageSource = texture.value(
+                    "/extensions/KHR_texture_basisu/source"_json_pointer, -1);
+                if (ktxImageSource == imgIndex) {
+                  texture["extras"] = json::object();
+                  texture["extras"]["ANARISampler"] = samplerExtras;
+                }
+              }
+
+            } else {
+              if (needsTranscoding) {
+                result = ktxTexture2_TranscodeBasis(texture, KTX_TTF_RGBA32, 0);
+                if (result != KTX_SUCCESS) {
+                  printf("Failed to load image %zu: %s\n", imgIndex, ktxErrorString(result));
+                }
+              }
+
+              *width = texture->baseWidth;
+              *height = texture->baseHeight;
+              *n = ktxTexture2_GetNumComponents(texture);
+              int fmt = texture->vkFormat;
+
+              result = ktxTexture_GetImageOffset(
+                  ktxTexture(texture), 0, 0, 0, &offset);
+
+              if (result != KTX_SUCCESS) {
+                printf("Failed to load image %zu: %s\n", imgIndex, ktxErrorString(result));
+              }
+              data = ktxTexture_GetData(ktxTexture(texture)) + offset;
+            }
+            ANARIMemoryDeleter d = [](const void *usrPtr, const void *mem) {
+              void *p = const_cast<void *>(usrPtr); 
+              ktxTexture2_Destroy(reinterpret_cast<ktxTexture2*>(p));
```

Comment:
I get an compile error here when using the latest release v4.3.2 of KTX, because of https://github.com/KhronosGroup/KTX-Software/issues/911.
How about
```cpp
ktxTexture_Destroy(reinterpret_cast<ktxTexture*>(p))
```
which should still call the correct virtual destructor.